### DISCUSSION TITLE
Translated slugs

### DISCUSSION THREAD

**Discussion by VojtaSim** [swellstores/community#542] [Help] [answered]
Hello,

Swell's internationalization support is great but I'd like to know how I can get product and category slugs to be translated as well...

My end goal is to have fully translated URLs on the frontend (NextJS) - `/en/products/<product-slug-en>` vs. `/cs/produkty/<product-slug-cs>` and fetch the product data with `productBySlug` GQL query.

I suppose custom fields won't get me there or I'd have to use `where` conditions which I can imagine isn't as optimized as `productBySlug` query.

**Comment by ericingram** [comment, marked as answer]
It's not possible yet, without using a `where` clause as you mentioned. It is definitely something we'd like to do in the future, as well as supporting redirects from slugs if they are changed.

That said, if you have less than a few thousand products then it would still be relatively performant to use a custom field.
